src/murmur/Messages.cpp

```diff
@@ -462,8 +464,27 @@ void Server::msgBanList(ServerUser *uSource, MumbleProto::BanList &msg) {
 				b.qdtStart = QDateTime::currentDateTime().toUTC();
 			}
 			b.iDuration = be.duration();
-			if (b.isValid())
+			if (b.isValid()) {
 				qlBans << b;
+				if (! tmpBans.contains(b)) {
+					log(uSource, QString("New or changed ban: %1, Host: %2, Username: %3, Reason: %4, Length: %5").arg(
+						b.qsHash,
+						b.haAddress.toString(),
+						b.qsUsername,
+						b.qsReason,
+						b.iDuration == 0 ? "Permanent" : "Temporary"));
```

Comment:
Given the potentially controversial nature of bans I think it might be helpful to log the whole actual ban with all fields. Because of that we should probably log  ```iMask``` and the given start and duration instead of trying to simplify it to permanent and temporary (though that might be useful as an addon note to simplify grepping for it). Would also add a "Hash: " before ```%1``` to make it clear it's the hash field (can be empty after all). Reason (maybe other fields too?) should be quoted.